### DISCUSSION TITLE
Random series of Disconnects

### DISCUSSION THREAD

**Discussion by Evolve6996** [vpnhood/VpnHood#82] [Q&A]
hello when I built my server with vpnhood I tested it and it had some ping spikes to 600 and there was packet loss **only on UDP mode**
but this wasn't that bad

recently I got random series of packet loss and disconnects on TCP and vpnhood will become unusable for a while and after that it back to normal condition other VPN from the same server which is just a regular OpenVPN working fine though

we have a firewall here in the country but I don't know if that trying to disrupt the connection since OpenVPN works fine and usually, firewall disruptions come after a while or right after connecting to VPN but this thing is really random also during this packet loss I can't even access to my local router!

my friend which uses the same vpnhood server has the same exact problem and he is getting service from another isp

----------------------------------------------------

Update 1: (for explaining condition more): sometimes during this vpnhood will stuck on the blue color startup page for like 40 sec and every time it happens my internet speed falls down to about 20 to 30-kilobyte p/s and there be a lot of packet loss and ping spike, which is pretty normal when your internet bandwidth suppressed like that.

-----------------------------------------------------------------
Update 2: (I am going to update the first post in case maybe this information is useful in the future for investigation also changed the name of discussion from Random series of Packet loss to Random series of Disconnects for being more related to the problem):

finally, Use UDP-mode started working for some unknown reason! and I tested it, it has the same problem as non-UDP mode. it just completely disconnects and reconnects after a while. it seems during this time not a single packet can transfer; in the log I see 
( | Udp | Packet sent to Tunnel queue ) but there are no Packets received from Tunnel message in the log at that particular time.
also, I found in there are some disconnects in OpenVPN over Cloak connection but maybe not frequent as vpnhood (maybe my VPS is problematic) 

--------------------------------------------

Update 3: Today I changed my hosting provider and installed vpnhood on new VPS in case the VPS might be problematic; but result was the same still disconnecting.

**I will tag and write about the fix and mark the discussion as answered when I found it solved.**

**Comment by bobvhood** [comment]
Yes, we have found serious UDP packet loss flaws in both server and client with high-speed connections. It is already fixed but we haven't released it, please wait for the next release (perhaps by next week), and let see the result. Monitor issue #83
Thank you for sharing the issue

**Reply by Evolve6996** [reply]
thanks for reply
I am currently running clients on No UDP mode is the issue you said still affecting my problem ?

**Reply by bobvhood** [reply]
Yes, the problem affected both UDP and no UDP mode. I hope the next version solves the issue you reported.

**Comment by trudyhood** [comment]
Dear @Evolve6996 
We have optimized and significantly improve VpnHoodServer stability, memory & CPU usage over high load servers.
Please check out the new version (192). We would be grateful if you let us know the result, especially for the UDP packet loss.

**Reply by Evolve6996** [reply]
thanks, I would do that and I also called my friend to test it I will report if there is anything since it's a random problem need to test for like a week
if it's possible please add killswitch faster ❤️ 👍

**Reply by trudyhood** [reply]
What do you mean by killswitch?

**Reply by Evolve6996** [reply]
> What do you mean by killswitch?

https://github.com/vpnhood/VpnHood/issues/53


> especially for the UDP packet loss.

I don't know what is your exact meaning by that I had two types of packet loss one is the one I described here was happening even in no UDP mode that needs more testing as I said

the other type is when i switch to use UDP mode and its now in this version is just like before and vpn is not working in this mode

**Reply by trudyhood** [reply]
OK, I've understood about killswitch
---
non-UDP Mode
In this mode, VpnHood sends UDP packets via some TCP connections. I am interested to see your testing about non-UDP mode packet loss. Please share the result after you test it.

UDP Mode
In this mode, VpnHood encapsulates UDP packets in another UDP packet as a payload. It works here, If it doesn't work at all for you or it works just for a few minutes, it means your internet provider is completely dropping unknown UDP packets and you can't use UDP protocol. That's why we have created the non-UDP mode. You shouldn't use UDP mode If your ISP actively limits your internet, because after transferring a few traffic, their system may block the connections as they don't expect huge traffic over UDP packets.

Feel free to share your mind.

**Reply by Evolve6996** [reply]
can you explain what is your meaning by term unknown UDP packets? 

we can play video games here and most of games netcodes running on UDP protocol these days
so I don't think they blocked it entirely.
but I can't even transfer 32bytes of ICMP ping packet in UDP mode and can't even watch 360p youtube I don't think these are huge traffic that's weird

**Reply by trudyhood** [reply]
Yes, many games, especially real-time use UDP, because it is faster and doesn't have TCP overload, but mostly they are small packets. If you switch to non-UDP mode, you may encounter lag in online games. Also, if you switch to UDP mode, all traffic will go through UDP, and your ISP may suspect the traffic. 

> but I can't even transfer 32bytes of ICMP ping packet in UDP mode and can't even watch 360p youtube. I don't think these are huge traffic that's weird.
Ya, it is so weird that you can't ping even via UDP mode. I am so intrigued to know the reason. 
1) Would you send us the diagnose? 
2) Also, please make sure your client firewall doesn't block UDP.
3) Do you have the same issue with OpenVPN in UDP mode?

**Reply by Evolve6996** [reply]
>doesn't have TCP overload, but mostly they are small packets. If you switch to non-UDP mode, you may encounter lag in online games.

I don't know what is the meaning of overload you say I disabled TCP ACK in windows client registry I don't know if it solves that overload


>Would you send us the diagnose?



>Also, please make sure your client firewall doesn't block UDP.

it makes no change

>Do you have the same issue with OpenVPN in UDP mode?

Ohhh basically, it's confusing to explain and deal with for me:

Regular OpenVPN connections from the VPS iam using only work on port 1194 UDP which is the default OpenVPN port. it's not working on other ports as I tested neither TCP nor UDP.

OpenVPN+Cloak works on both TCP and UDP with port 1194.

other OpenVPN connections (which I don't know if they are regular ones or not) from big vpns like Express-vpn and freedome-vpn which are set on ports 1195 and 2745 UDP; are just fine for my internet connection.

**Reply by trudyhood** [reply]
1) Sorry, I meant TCP overhead, but it is inevitable. Also, disabling TCP ACK sounds crazy to me!
2) I checked the diagnose. You are still using the old version of VpnHood server. You should use version 192 for both server and client
3) Please let me know can you connect to our Public Server via UDP and send a few packets like ping?
4) Are you sure that your Server Firewall configuration allows any UDP ports? VpnHood assigns a random UDP port, and your server must not block any UDP port. Maybe you have missed to configure network firewall offered by your hosting (not the one in the OS).

**Reply by Evolve6996** [reply]
>Sorry, I meant TCP overhead, but it is inevitable. Also, disabling TCP ACK sounds crazy to me!

sorry if this is an offtopic question do you think it might corrupt my browser data sometimes like HTTPS data?

>I checked the diagnose. You are still using the old version of the VpnHood server. You should use version 192 for both server and client

oh, I thought it will update automatically update by itself how can I update it without causing problems with my access keys?

>Please let me know can you connect to our Public Server via UDP and send a few packets like ping?

its same as my server I cant do ping

>Are you sure that your Server Firewall configuration allows any UDP ports? VpnHood assigns a random UDP port, and your server must not block any UDP port. Maybe you have missed configuring the network firewall offered by your hosting (not the one in the OS).

since the public vpnhood server has the same result I don't think its the case

**Reply by trudyhood** [reply]
>oh, I thought it will update automatically update by itself how can I update it without causing problems with my access keys?
It should get the update unless you remove the update configuration or maybe there is a bug in autoupdate. If you are using our access server, you don't need to worry about losing your access keys otherwise just make sure to have backup of the tokens folders

> its same as my server I cant do ping
Maybe you can just send or receive from some specifics UDP port such as 1195. Can you check and make sure you can send/receive UDP packets via other ports to your server. You can use nping, or any other tools that you are already familiar with.
https://nmap.org/nping/

**Reply by Evolve6996** [reply]
ok for some reason it seems the server updated only for one version to 283 and stuck there btw I updated it manually also i am not sure why in default appsetting.json the default port was 1443 instead of 443


>Maybe you can just send or receive from some specifics UDP port such as 1195. Can you check and make sure you can send/receive UDP packets via other ports to your server. You can use nping, or any other tools that you are already familiar with.
https://nmap.org/nping/

it seems very confusing to me. btw as you said I ran an echo server by nping here is the result :
there is two-port 1195(UDP) and the second is vpnhood access key port which is 3389(TCP)

**Reply by trudyhood** [reply]
As you see you have 100% udp packet loss. I also couldn't udp ping from one of my Server. Are you sure you run udp nping server? would you keep it running? I need to ping it too via udp.

**Reply by Evolve6996** [reply]
its running now i dont know how to run nping server in udp mode

**Reply by trudyhood** [reply]
nping --ec 123456 136.243.214.125 --udp -p1195
Your server does not reply to my server.  If the ping server is running properly, It means the firewall of your server OS or your network firewall blocks UDP. You need to go to your hosting panel and make sure all UDP ports are open.

**Reply by trudyhood** [reply]
Also, I see your server reply TCP. We are testing UDP. Make sure you have run ping server for UDP not TCP

**Reply by Evolve6996** [reply]
the command you said is the same command as what I showed in the picture
I don't know how to run nping echo server in UDP mode and couldn't find the way on the internet

**Reply by trudyhood** [reply]
run this command on your server:
nping --es 123456 --udp -p15500

**Reply by Evolve6996** [reply]
it's running now if you wanna test yourself it had the same result as before for me all packets lost

>You need to go to your hosting panel and make sure all UDP ports are open

i don't have access to this option in my host panel and when vpnhood public server is also giving me 100% packet loss on udp i don't think it would change the condition if I open the port on the host panel

the last assumption is government firewall blocked UDP which again is not the case, cause OpenVPN from other big VPNs likes expressvpn working ok on udp here

**Reply by trudyhood** [reply]
Let find the problem one by one. Find why I can not ping your server via UDP from a server in US despite your government firewall!?
I tried both port:
 nping --ec 123456 136.243.214.125 --udp -p1195
 nping --ec 123456 136.243.214.125 --udp p15500

**Reply by Evolve6996** [reply]
I have to find and really invest more time on it in feature. also, I have other types of VPNs and configs maybe they are problematic and for the final idea, I could say probably the mixture of host firewall and government intelligent firewall caused all of this.
there are other clues though I can only run OpenVPN on UDP port 1194 and previously just in a one of my attempts for running vpns in this host I wasted 4 days to run local bridge and tap interface for softether VPN and finally, it didn't work maybe there is a relation 😢 

btw thanks for your efforts Trudy ❤️ I will report if the random TCP packet loss is fixed in the new version of vpnhood cause I currently gave up on UDP

**Reply by trudyhood** [reply]
Don't give up. Your problem still is not the government firewall (at least now), because I see your server is in Germany and I can not ping it from the US. Fix it first. Disable your server firewall and contact your hosting company. This issue must be solved.

You can run OpenVPN on 1194 UDP because usually, this port is open by firewalls as default.

**Reply by Evolve6996** [reply]
hello again
unfortunately, my problem is not solved still even in TCP mode. I don't know if its 100 % vpnhood but I updated and explained the condition with more details in first post and will share what I have found in terms of some sort of workaround fix for this

Fix: it seems turnning off and on android wifi will solve it for a while.

**Reply by bobvhood** [reply]
Hi, Does your IP changes when you restart your wifi? Do you have the same issue with windows?

**Reply by Evolve6996** [reply]
>Hi, Does your IP changes when you restart your wifi?

i didn't say i restart my wifi

>Do you have the same issue with windows?

due to some limitations i don't use vpnhood on pc

**Reply by bobvhood** [reply]
I have an idea, we need to make sure what happens when you start losing packets. 
Can you check your TCP connections at that time?
When you browse a website it usually send a UDP packet to resolve DNS, so let's try to open a website with IP address

1. Connect your Vpnhood 
2. Work till you start packet loss and problem
3. open this address: http://13.226.210.103/
4. I expect you to see the following error:
![image](https://user-images.githubusercontent.com/92276967/149585133-8387d320-d1c9-471f-bc36-a4622d501917.png)

It helps us to diagnose whether the issue is caused by our datagram channel or something else. 
* Make sure to refresh the page so it does not come from the cache
* Try to browse other website to see others work or not
* All test should be done while you have connected a VpnHood server

**Reply by Evolve6996** [reply]
>Can you check your TCP connections at that time?
how can i do that in an android phone and what should i seek for?

>open this address: http://13.226.210.103/
will report when i opened it

**Reply by bobvhood** [reply]
Just wanna make sure: 
you have tried to open this address: http://13.226.210.103/ after connecting to VpnHood and at the same time and you couldn't open any other site? If yes, please start in diagnose mode and try it again, then try to connect to a website that doesn't work, disconnect and send us the report.

**Reply by Evolve6996** [reply]
> open this address: http://13.226.210.103/
I can open it

it's kinda hard to get a diagnose since the app is crashing during that but I got something

**Reply by trudyhood** [reply]
Hi, I checked the log file. Unfortunately, there is not much information for us to understand the issue. By the way, can we have access to your server? Even a non-root user would be enough. If so, we can log in to your server, install an instance of VpnHood and make sure there is no issue or may debug any potential problem. If you are interested, email your server credential to me support@vpnhood.com

**Reply by Evolve6996** [reply]
thanks 👍 sent an user to email

**Reply by trudyhood** [reply]
Ok, Thanks, right now we have to apply an update to the access server. I will check your server after the access server gets reliable.
